packages/babel-core/src/config/plugin.js

```diff
@@ -2,8 +2,8 @@
 
 export default class Plugin {
   constructor(plugin: {}, key?: string) {
-    if (plugin.name != null && typeof plugin.name !== "string") {
-      throw new Error("Plugin .name must be a string, null, or undefined");
+    if (!plugin.name || typeof plugin.name !== "string") {
```

Comment:
yep @loganfsmyth, this was discussed in the meeting on [7 June](https://github.com/babel/notes/blob/master/2017-06/june-07.md) (altho I wasn't around).

It has to do with simplifying the "intelligent plugin sorting" process. 

#### Benefits
1. Allows for global topological ordering with the DAG map.
2. Serves as a known identifying key for the plugin.
3. Provides opportunity to deduplicate identically configured plugins.

#### Steps
1. Deprecate unnamed plugins in Babel 6.
2. Add deprecation message.
3. Publish a new patch version with that deprecation.
4. Write a deprecation guide on how to make this change for your plugins.
5. Enforce named plugins in Babel 7.

Source: https://docs.google.com/document/d/1zO1QiaAyDxmA1OcmdOKYLFmuSXGSr4S50pE0uQFTVEE/edit#heading=h.bcrawxu2t468.